garb/garb_main.cpp

```diff
@@ -20,21 +20,30 @@
 
 void my_write_libcoredumper(int sig, const char *path, time_t curr_time) {
   int ret = 0;
-  char suffix[512];
-  char core[512];
+  static constexpr std::size_t buf_size = 512;
+  char suffix[buf_size];
+  char core[buf_size];
+  memset(suffix, '\0', buf_size);
+  memset(core, '\0', buf_size);
   struct tm *timeinfo = gmtime(&curr_time);
-  if (path == NULL)
+
+  if (path == nullptr)
     strcpy(core, "core");
   else
-    strcpy(core, path);
+    strncpy(core, path, buf_size);
```

Comment:
buf_size-1
or
core[buf_size=1] = 0 after this strncpy
We need trailing zero in case of path being longer than buf_size

Reply:
Fixed.